### PR TITLE
add example for gin instrumentation

### PR DESCRIPTION
**Pull request by andriikushch**
Is a part of https://github.com/instana/go-sensor/pull/231. Contains examples of gin instrumentation.

### CODE REVIEW

**Comment by andriikushch** [comment]
JFYI @andrewslotin

File: example/gin/go.mod [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module github.com/instana/go-sensor/example/gin
+
+go 1.11
+
+require (
+	github.com/gin-gonic/gin v1.7.2
+	github.com/instana/go-sensor v1.29.1-0.20210608074124-c12a7d59b2d4
+	github.com/instana/go-sensor/instrumentation/instagin v0.0.0-20210608075151-0d8105436b0c // indirect
```
Comment (andrewslotin) [inline example/gin/go.mod:8]: We'd probably need to merge and release https://github.com/instana/go-sensor/pull/234 before this one, right?

**Review by andrewslotin — COMMENTED** [review]
Overall LGTM, let's merge it as soon as https://github.com/instana/go-sensor/pull/234 is done

File: example/gin/go.mod (hunk 1 above)
Comment (andriikushch) [reply example/gin/go.mod:8]: Yes, it is a plan

**Pull request merged**